packages/react/tooltip/src/Tooltip.tsx

```diff
@@ -61,109 +128,76 @@ interface TooltipProps {
   onOpenChange?: (open: boolean) => void;
 
   /**
-   * The duration from when the mouse enters the trigger until the tooltip gets opened.
-   * (default: 700)
+   * The duration from when the mouse enters the trigger until the tooltip gets opened. This will
+   * override the prop with the same name passed to Provider.
+   * @defaultValue 700
    */
   delayDuration?: number;
-
-  /**
-   * How much time a user has to enter another trigger without incurring a delay again.
-   * (default: 300)
-   */
-  skipDelayDuration?: number;
   children?: React.ReactNode;
 }
 
 const Tooltip: React.FC<TooltipProps> = (props: ScopedProps<TooltipProps>) => {
   const {
-    __scopeTooltip,
+    __scopeTooltipProvider,
     children,
     open: openProp,
     defaultOpen = false,
     onOpenChange,
-    delayDuration = 700,
-    skipDelayDuration = 300,
+    delayDuration: delayDurationProp,
   } = props;
-  const popperScope = usePopperScope(__scopeTooltip);
+  const context = useTooltipProviderContext(TOOLTIP_NAME, __scopeTooltipProvider);
+  const popperScope = usePopperScope(__scopeTooltipProvider);
   const [trigger, setTrigger] = React.useState<HTMLButtonElement | null>(null);
   const contentId = useId();
+  const openTimerRef = React.useRef(0);
+  const delayDuration = delayDurationProp ?? context.delayDuration;
+  const openDelay = context.isOpenDelayed ? delayDuration : 0;
+  const wasOpenDelayed = usePrevious(Boolean(openDelay));
+  const { onTooltipOpen, onTooltipClose } = context;
   const [open = false, setOpen] = useControllableState({
     prop: openProp,
     defaultProp: defaultOpen,
     onChange: onOpenChange,
   });
-  const [stateAttribute, setStateAttribute] = React.useState<StateAttribute>(
-    openProp ? 'instant-open' : 'closed'
-  );
 
-  // control open state using state machine subscription
-  React.useEffect(() => {
-    const unsubscribe = stateMachine.subscribe(({ state, context }) => {
-      if (state === 'open' && context.id === contentId) {
-        setOpen(true);
-      } else {
-        setOpen(false);
-      }
-    });
+  const stateAttribute = React.useMemo(() => {
+    return open ? (wasOpenDelayed ? 'delayed-open' : 'instant-open') : 'closed';
+  }, [wasOpenDelayed, open]);
 
-    return unsubscribe;
-  }, [contentId, setOpen]);
-
-  // sync state attribute with using state machine subscription
-  React.useEffect(() => {
-    const unsubscribe = stateMachine.subscribe(({ state, context }) => {
-      if (context.id === contentId) {
-        if (state === 'open') {
-          setStateAttribute(context.delayed ? 'delayed-open' : 'instant-open');
-        } else {
-          setStateAttribute('closed');
-        }
-      } else {
-        setStateAttribute('closed');
-      }
+  const handleOpen = React.useCallback(() => {
+    setOpen((prevOpen) => {
+      if (!prevOpen) document.dispatchEvent(new CustomEvent(TOOLTIP_OPEN));
+      return true;
     });
+    onTooltipOpen();
```

Comment:
I'm calling `onTooltipOpen` here and `onTooltipClose` further down, regardless of whether the consumer prevents open/close when controlled. I tested how it behaves in that case and obviously the `isOpenDelayed` will change even if it wasn't opened so moving to the next tooltip will instantly open. 

I tried a fix for this with a `React.useEffect` that runs when `open` changes and then fires `onTooltipOpen` / `onTooltipClose`. However, that was too late to cancel the timers in some cases. I didn't invest too much time trying to figure this all out as it feels like an extreme edge-case. My thinking is that nothing breaks and if anyone raises it we can look into it? Thoughts?

Reply:
Oh, I just tested this exact same use-case in the previous version and it has the same issue so I guess it's clear by now that no one has noticed it except me 😅